crates/topos-sequencer-subnet-runtime/tests/subnet_contract.rs

```diff
@@ -921,3 +921,98 @@ async fn test_subnet_send_token_processing(
     context.shutdown().await?;
     Ok(())
 }
+
+/// Test sync of blocks and generating certificates from genesis block
+#[rstest]
+#[test(tokio::test)]
+#[timeout(std::time::Duration::from_secs(600))]
+#[serial]
+async fn test_sync_from_genesis(
```

Comment:
Here we have
- should start creating from the first block if the source head fetched from the TCE is `0` or non-existant

Do we have also one test for the general case?
- should start creating from `N+1` if the source head fetched from the TCE is `N`

Reply:
I'll expand the test to test for start from N+1

Reply:
The test is now like >200 lines while it feels better suited to be separated in two tests instead no?